aikau/src/main/resources/alfresco/forms/controls/MultiSelect.js

```diff
@@ -0,0 +1,1230 @@
+          *
+          * @instance
+          * @returns {string[]} The value(s) of the control
+          */
+         getValue: function alfresco_forms_controls_MultiSelect__getValue() {
+            return this.value;
+         },
+
+         /**
+          * Set the value of the control
+          *
+          * @instance
+          * @param    {string|string[]|object|object[]} newValue The new value(s)
+          */
+         setValue: function alfresco_forms_controls_MultiSelect__setValue(newValueParam) {
+
+            // Setup helper vars
+            var labelAttrName = this.store.labelAttribute,
+               valueAttrName = this.store.valueAttribute,
+               newValuesArray = newValueParam;
+            if (!ObjectTypeUtils.isArray(newValuesArray)) {
+               newValuesArray = (newValueParam && [newValueParam]) || [];
+            }
+
+            // Create an items array
+            var chosenItems = array.map(newValuesArray, function(nextNewValue) {
+
+               // Try and get the value from the existing result items
+               var newValueIsString = typeof nextNewValue === "string",
+                  realValue = (newValueIsString && nextNewValue) || nextNewValue[valueAttrName],
+                  nextItem = this._storeItems[realValue];
+
+               // If we already have the item, return immediately
+               if (nextItem) {
+                  return nextItem;
+               }
+
+               // Use value if object, otherwise add as value property to new object
+               if (newValueIsString) {
+                  nextItem = {};
+                  nextItem[valueAttrName] = realValue;
+               } else {
+                  nextItem = nextNewValue;
+               }
+
+               // Add a temporary label if not present
+               if (!nextItem.hasOwnProperty(labelAttrName)) {
+                  nextItem[labelAttrName] = nextItem[valueAttrName];
+               }
+
+               // Put the new item into the items map
+               this._storeItems[realValue] = nextItem;
+
+               // Mark this item as needing updating from the store
+               this._itemsToUpdateFromStore.push(nextItem);
+
+               // Pass back the final item
+               return nextItem;
+
+            }, this);
+
+            // Add the choices to the control and kick off the label retrieval if necessary
+            array.forEach(chosenItems, function(nextItem) {
+               var label = nextItem[labelAttrName],
+                  value = nextItem[valueAttrName];
+               this._addChoice(label, value);
+            }, this);
+            this._updateItemsFromStore();
+            this._updateResultsDropdown();
+
+            // Set the new value
+            this._changeAttrValue("value", chosenItems);
+         },
+
+         /**
+          * Add the specified result item to the choices
+          *
+          * @instance
+          * @protected
```

Comment:
We need to be a bit careful about how we use JSDoc attributes like this as we're looking to somewhat abuse them to define what is and isn't safe from an extensibility point of view, this is still up for discussion so for the time being I'd like to avoid us using them to prevent possible confusion in the future